### PR TITLE
feat: add paypal and paypalpaylater additional data on connection options

### PR DESCRIPTION
**Pull request by galbino**
# Description
Added `additional_data` parameter to `paypal-paypal` and `paypal-paypalpaylater` on connection options

Implements TA-4063

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own changes
- [X] I have run `yarn test` to make sure my changes pass all linters
- [X] I have pulled the latest changes from the upstream `main` branch

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

File: reference/resources/connection_options.yaml [hunk 1]
```diff
@@ -64,3 +64,49 @@ properties:
...
+    nullable: true
+    description: |-
+      Additional options to be passed through to Paypal when processing
+      transactions.
+    properties:
+      additional_data:
+        type: array
+        description: |-
+          An array with key-value objects representing additional
+          data to be passed to Paypal.
```
Comment (cbetta) [inline reference/resources/connection_options.yaml:79]: @galbino What field on the PayPal side is this mapped to? Is it called "additional_data" on the PayPal API?

File: .spelling [hunk 2]
```diff
@@ -23,6 +23,7 @@ vh
 vw
 timestamp
 Cybersource
+Paypal
```
Comment (cbetta) [inline .spelling:26]: It's PayPal, not Paypal.

**Review by cbetta — CHANGES_REQUESTED** [review]
Please rename Paypal to PayPal

File: reference/resources/connection_options.yaml (hunk 1 above)
Comment (galbino) [reply reference/resources/connection_options.yaml]: yep, thats right

Comment (galbino) [reply reference/resources/connection_options.yaml]: @cbetta 
image

**Review by cbetta — APPROVED** [review]

**Pull request merged**